### PR TITLE
Edit bale

### PR DESCRIPTION
**Pull request by gastonabella**
### Brief Description
Edit bales service ready for frontend.
### Related card or ticket
https://github.com/Pis-moove-it/reciclando-backend/issues/8
### Additional Details
- Edit bales service
- Controller tests
- Swagger (with an improvement of POST method)
- Rubocop configured in order to allow an 80-lines-block (necessary for controller test).

### CODE REVIEW

File: app/controllers/bales_controller.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 class BalesController < BaseController
+  protect_from_forgery with: :null_session
```
Comment (fbalarini) [inline app/controllers/bales_controller.rb]: We should add this in `Base controller`

File: app/controllers/bales_controller.rb [hunk 2]
```diff
@@ -16,6 +17,14 @@ def show
...
+    if bale_by_id.update(bale_params)
+      head :ok
+    else
+      render_error(1, bale_by_id.errors)
+    end
+  end
+
   private
 
   def bale_by_id
```
Comment (fbalarini) [inline app/controllers/bales_controller.rb]: What do you think about renaming this method as `bale` ? I think it is shorter and the controller is more clear this way

File: spec/controllers/bales_controller_spec.rb [hunk 3]
```diff
@@ -61,4 +62,29 @@ def bale_by_id_call(id)
...
+      it 'should return success' do
+        update_bale_call(bale.id, updated_bale.weight, updated_bale.material)
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    context 'when a failure response is expected' do
+      it 'should return not found' do
+        update_bale_call(Bale.pluck(:id).max + 1, updated_bale.weight, updated_bale.material)
+        expect(response).to have_http_status(:not_found)
```
Comment (fbalarini) [inline spec/controllers/bales_controller_spec.rb]: test we are returning the right error response

File: spec/controllers/bales_controller_spec.rb [hunk 4]
```diff
@@ -61,4 +62,29 @@ def bale_by_id_call(id)
...
+      end
+
+      it 'should not update a bale because it has empty attributes' do
+        update_bale_call(bale.id, nil, nil)
+        expect(response).to have_http_status(:bad_request)
```
Comment (fbalarini) [inline spec/controllers/bales_controller_spec.rb]: same here

File: spec/controllers/bales_controller_spec.rb [hunk 5]
```diff
@@ -57,7 +58,7 @@ def bale_by_id_call(id)
 
     context 'Failure cases' do
       it 'should return not found' do
-        bale_by_id_call(Bale.pluck(:id).max + 1)
+        bale_by_id_call(invalid_id)
```
Comment (fbalarini) [inline spec/controllers/bales_controller_spec.rb:61]: I think it would be better to check all in the same test

**Review by fbalarini — APPROVED** [review]

File: spec/controllers/bales_controller_spec.rb (hunk 5 above)
Comment (fbalarini) [reply spec/controllers/bales_controller_spec.rb:61]: btw, checking the error code from errors.yaml is enough

**Comment by accg14** [comment]
Please resolve conflicts, and check out rubocop

**Pull request merged**